unix/unix_cstruct.ml

```diff
@@ -17,3 +17,56 @@
 let of_fd fd =
   let buffer = Bigarray.(array1_of_genarray (Unix.map_file fd char c_layout false [|-1|])) in
   Cstruct.of_bigarray buffer
+
+(* Returns 0 if there is no writev *)
+external stub_iov_max: unit -> int = "stub_cstruct_iov_max"
+
+external stub_write: Unix.file_descr -> Cstruct.t -> int = "stub_cstruct_write"
+
+external stub_writev: Unix.file_descr -> Cstruct.t list -> int = "stub_cstruct_writev"
+
+let iov_max = stub_iov_max ()
+
+(* return the first n fragments, suitable for writev *)
+let rec first n rev_acc = function
```

Comment:
I mildly dislike having to do these potentially O(n) list operations for a writev, and mandate that in the external interface.   In your vpnkit usecase, would you be able to pass in a `Cstruct.t array` instead for writev, and then just call writev with simple O(1) time slices instead?

It affects the interface we expose in the mli, hence my asking now (as opposed to optimising later)